validator/client/validator.go

```diff
@@ -1055,15 +1053,34 @@ func (v *validator) buildProposerSettingsRequests(ctx context.Context, pubkeys [
 			})
 		}
 		if !skipAppendToFeeRecipientArray && enableValidatorRegistration {
-			registerValidatorRequests = append(registerValidatorRequests, &ethpb.ValidatorRegistrationV1{
-				FeeRecipient: feeRecipient[:],
-				GasLimit:     gasLimit,
-				Timestamp:    uint64(time.Now().UTC().Unix()),
-				Pubkey:       pubkeys[i][:],
-			})
+			signedReg, ok := v.signedValidatorRegistrationCache[pubkeys[i]]
```

Comment:
We probably want locks around the cache if we foresee concurrent access. Dynamic update in the future?

Reply:
I don't think this field will be updated concurrently. it should only run once every epoch, but since you mentioned this maybe I need to update the map that stores the proposer settings and add a lock on that.